hal/src/main/native/sim/HAL.cpp

```diff
@@ -193,7 +201,68 @@ HAL_Bool HAL_GetBrownedOut(int32_t* status) {
   return false;  // Figure out if we need to detect a brownout condition
 }
 
+/* A return < 0 indicates an error that should stop the HAL.
+   0 is success, and > 0 is a non fatal error */
+typedef int halsim_init_func_t(void);
+
+int HAL_LoadExtraSimulation(const char* library) {
+  int rc = 1;  // It is expected and reasonable not to find an extra simulation
+#if defined(WIN32) || defined(_WIN32)
+  HMODULE handle = LoadLibrary(library);
+  if (!handle) {
+    std::string library_name = std::string(library) + ".dll";
```

Comment:
Please use llvm::SmallString for this type of thing.

Reply:
`LoadLibrary` does not require the `.dll` on `<library>.dll`. Just `<library>` is used.